lost-and-found-service-persistence/src/test/java/cz/muni/fi/CategoryDaoImplTest.java

```diff
@@ -2,129 +2,115 @@
-        assertThatThrownBy(() -> categoryDao.updateCategory(electro)).hasCauseInstanceOf(IllegalArgumentException.class);
+        entityManager.persist(electro);
+        electro.setId(null);
+        categoryDao.updateCategory(electro);
     }
 
-    @Test
+    @Test(expectedExceptions = IllegalArgumentException.class)
     public void deleteNullCategory() {
-        assertThatThrownBy(() -> categoryDao.deleteCategory(null)).hasCauseInstanceOf(IllegalArgumentException.class);
+        categoryDao.deleteCategory(null);
     }
 
-    @Test
+    @Test(expectedExceptions = IllegalArgumentException.class)
     public void deleteNullIdCategory() {
-        assertThatThrownBy(() -> categoryDao.deleteCategory(electro)).hasCauseInstanceOf(IllegalArgumentException.class);
+        entityManager.persist(electro);
+        electro.setId(null);
+        categoryDao.deleteCategory(electro);
     }
 
-    @Test
+    @Test(expectedExceptions = IllegalArgumentException.class)
     public void getNullCategoryById() {
-        assertThatThrownBy(() -> categoryDao.getCategoryById(null)).hasCauseInstanceOf(IllegalArgumentException.class);
+        categoryDao.getCategoryById(null);
     }
 
     @Test
-    public void shouldAddCategory() {
+    public void testAddCategory() {
         categoryDao.addCategory(electro);
-        assertEquals(categoryDao.getAllCategories().size(), 1);
+        assertEquals(electro, entityManager.find(Category.class, electro.getId()));
     }
 
     @Test
-    public void shouldDeleteCategory() {
-        categoryDao.addCategory(electro);
-        categoryDao.addCategory(clothes);
-        assertEquals(categoryDao.getAllCategories().size(), 2);
+    public void testDeleteCategory() {
+        entityManager.persist(electro);
         categoryDao.deleteCategory(electro);
-        assertEquals(categoryDao.getAllCategories().size(), 1);
-
-        assertNull(categoryDao.getCategoryById(electro.getId()));
-        assertEquals(categoryDao.getCategoryById(clothes.getId()), clothes);
+        assertNull(entityManager.find(Category.class, electro.getId()));
     }
 
     @Test
-    public void shouldUpdateCategory() {
-        categoryDao.addCategory(electro);
-        String newName = "Electronics";
-
-        electro.setName(newName);
-        categoryDao.updateCategory(electro);
-
-        Category updatedCategory = categoryDao.getCategoryById(electro.getId());
+    public void testUpdateCategory() {
+        entityManager.persist(electro);
+        Category modified = electro;
+        modified.setName("Electronics");
+        categoryDao.updateCategory(modified);
+        assertEquals(electro, modified);
 
-        assertEquals(updatedCategory.getName(), newName);
     }
 
     @Test
-    public void shouldReturn0CategoriesWhenEmpty() {
+    public void testGetAllCategoriesWhenEmpty() {
         assertEquals(categoryDao.getAllCategories().size(), 0);
-        assertNull(categoryDao.getCategoryById(0L));
     }
-*/
 
```

Comment:
Can you add test for getAllCategories() when categoryDao is not empty? This scenario is missing.